bridges/snowbridge/docs/v2.md

```diff
@@ -0,0 +1,310 @@
+# Snowbridge V2
+
+This design lowers fees, improves UX, improves relayer decentralization and allows "transacting" over the bridge, making it a general-purpose bridge rather than just a token bridge.
+
+We're grateful to Adrian Catangiu, Francisco Aguirre, and others from the Parity XCM/Bridges team for their help and collaboration on this design.
+
+# Summary
+
+- Unordered messaging
+- All messages routed through AH
+- Off-chain fee estimation
+- P→E Fee Asset: WETH
+- E→P Fee Asset: ETH
+- Relayer rewards for both directions paid out on AH in WETH
+
+# Polkadot→Ethereum
+
+Given source parachain $S$, with native token $S^{'}$ and the initial xcm $x_0$ to be executed on $S$.
+
+## Step 1: User agent constructs initial XCM
+
+The user agent constructs an initial XCM message $x_0$ that will be executed on S.
+
+The fee amounts in this message should be high enough to enable dry-running, after which they will be lowered.
+
+## Step 2: User agent estimates fees
+
+- Given source parachain $S$, with native token $S^{'}$ and the initial xcm $x_0$ to be executed on $S$.
+- The native currency $P^{'}$ of the Polkadot relay chain, and $E^{'}$ of Ethereum.
+- Suppose that the user agent chooses relayer reward $r$ in $E^{'}$.
+- Suppose that the exchange rates  $K_{P^{'}/S^{'}}$ and $K_{E^{'}/S^{'}}$. The user agent chooses a multiplier to $\beta$ to cover volatility in these rates.
+
+Apply the following sequence operations:
+
+1. Dry-run $x_0$ on $S$ to receive xcm $x_1$ and cost $a$ in $S^{'}$
+2. Dry-run $x_1$ on AH to receive xcm $x_2$ and cost $b$ in DOT
+3. Dry-run $x_2$ on BH to receive command $m$ and cost $c$ in DOT
+4. Dry-run $m$ on Ethereum to receive cost $d$ in ETH
+
+The final cost to the user in $S^{'}$ is given by
+
+$$
+\beta \left(a + \frac{b + c}{K_{P^{'}/S^{'}}} + \frac{d + r}{K_{E^{'}/S^{'}}}\right)
+$$
+
+The user agent should perform a final update to xcm $x_0$, substituting the calculated fee amounts.
+
+## Step 3: User agent initiates bridging operation
+
+The user agent calls `pallet_xcm::execute` with the initial xcm $x_0$
+
+```
+WithdrawAsset (KLT, 100)
+PayFees (KLT, 20)
+InitiateAssetsTransfer asset=(KLT, 60) remoteFee=(KLT, 20) dest=AH
+	ExchangeAsset give=(KLT, 20) want=(WETH, 1)
+	InitiateAssetsTransfer asset=(KLT, 40) remoteFee=(WETH, 1) dest=Ethereum
+		DepositAsset (KLT, 40) beneficiary=Bob
+```
+
+## Step 4:  AH executes message x1
+
+The message $x_1$ is application-specific:
+
+```
+ReserveAssetDeposited (KLT, 80)
+PayFees (KLT, 20)
+SetAssetClaimer Kilt/Alice
+AliasOrigin Kilt/Alice
+ExchangeAsset give=(KLT, 20) want=(WETH, 1)
+InitiateAssetsTransfer asset=(KLT, 60) remoteFee=(WETH, 1) dest=Ethereum
+	DepositAsset (KLT, 60) beneficiary=Bob
+```
+or
+```
+*ReserveAssetDeposited (KLT, 80)
+*PayFees (KLT, 20)
+*SetAssetClaimer Kilt/Alice
```

Comment:
I was confused at first with this, since after `AliasOrigin`, the claimer of the assets will be `Kilt/Alice`. However, I realized that `AliasOrigin` itself might fail, if the chain is not configured to accept that. In that case it makes sense to do `SetAssetClaimer` before with the same exact origin.

A little note would be helpful to others like me that got might get confused about it 😅